### PR TITLE
[SMALLFIX] Fix script location lookup when script is not a link

### PR DESCRIPTION
**Pull request by aaudiber**
If readlink fails, fall back to non-readlink

### CODE REVIEW

**Comment by aaudiber** [comment]
Merging now since I broke everything - please post-review

File: bin/alluxio [hunk 1]
```diff
@@ -181,7 +181,7 @@ function main {
   if [[ $- == *x* ]]; then
     LAUNCHER="bash -x"
   fi
-  BIN=$(cd "$( dirname "$( readlink "$0" )" )"; pwd)
+  BIN=$(cd "$( dirname "$( readlink "$0" || echo "$0" )" )"; pwd)
```
Comment (yupeng9) [inline bin/alluxio]: Shall we define this a function as it becomes complicated and used multiple times.

Comment (aaudiber) [inline bin/alluxio]: We'd need to put that function in a common file, but then we can't find the file without calling the function first x.x

Comment (yupeng9) [inline bin/alluxio]: yup. that's a bummer...

**Pull request merged**